docs/howto/charm-to-python.rst

```diff
@@ -0,0 +1,126 @@
+.. _howto-migrate-to-python:
+
+Migrate from the ``charm`` plugin to the ``python`` plugin
+==========================================================
+
+The ``python`` Charmcraft plugin offers a faster, stricter means of packing an operator
+charm with a virtual environment. This guide shows how to migrate from a charm using
+the default ``charm`` plugin to using the ``python`` plugin.
+
+Update ``charmcraft.yaml``
+--------------------------
+
+The first step is to update ``charmcraft.yaml`` to include the correct parts definition.
+Depending on the history of a specific charm, it may not have an explicitly-included
+``parts`` section determining how to build the charm. In this case, a ``parts`` section
+can be created as follows:
+
+.. code-block:: yaml
+
+    parts:
+      my-charm:  # This can be named anything you want
+        plugin: python
+        source: .
+        python-requirements:
+          - requirements.txt  # Or whatever your requirements file is called.
+
+Ensure a recent enough version of ``pip``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``python`` plugin requires at least pip 22.3, released in October 2022. If the
```

Comment:
> pip 22.3, released in October 2022

Rather than mentioning the date, can we link to https://pypi.org/project/pip/22.3 instead?

Reply:
We can link there as well, but the date is also useful because it gives them a good idea as to whether their base distribution is going to have a new enough pip out of the box. For example, CentOS 7 and Ubuntu Jammy won't because they were all released before then (so they'll need to follow this), but Ubuntu Noble does. I don't want to list specific bases because that just gives us a treadmill of bases to update.